src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,54 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: ($day-size + $gap) * $col-count - $gap;
+  gap: $gap;
+  padding: $padding;
+
+  &__day {
+    display: flex;
+    height: $day-size;
+    width: $day-size;
+    border: 1px solid black;
+    background-color: #eee;
+
+    position: relative;
+    text-align: center;
+    align-items: center;
+    justify-content: center;
+    box-sizing: border-box;
+
+    &:hover {
+      cursor: pointer;
+      background-color: #ffbfcb;
+      transform: translate(0, -20px);
+      transition-duration: 0.5s;
+    }
+
+    @for $number from 1 through $month-length {
+      &:nth-child(#{$number})::before {
+        content: '#{$number}';
+        display: block;
+        position: absolute;
+        font-family: Arial, Helvetica, sans-serif;
+        font-size: 30px;
+      }
+    }
+  }
+
+  @each $day, $index in $week-day {
```

Comment:
Ensure that the variable $week-day is defined as a list or map of days. This is necessary for the @each loop to function correctly.